### ISSUE TITLE
malformed __eh_frame section: too little space allocated for parseCFIs for architecture x86_64

### ISSUE THREAD

**Issue by desunit** [dotnet/coreclr#1594]
Hi,

I was trying to build CoreCLR with XCode 7 but received the following error:
```
$ ./build.sh
...
[ 27%] Linking CXX static library libstrongname_crossgen.a
Building CXX object src/vm/dac/CMakeFiles/cee_dac.dir/__/formattype.cpp.o
ld: in ../../pal/src/libcoreclrpal.a(dispatchexceptionwrapper.S.o), malformed __eh_frame section: too little space allocated for parseCFIs for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/dlls/dbgshim/libdbgshim.dylib] Error 1
make[1]: *** [src/dlls/dbgshim/CMakeFiles/dbgshim.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 27%] Building CXX object src/binder/v3binder_crossgen/CMakeFiles/v3binder_crossgen.dir/__/failurecache.cpp.o
[ 27%] Built target strongname_crossgen
...
[ 42%] Building ASM object src/vm/wks/CMakeFiles/cee_wks.dir/__/amd64/virtualcallstubamd64.S.o
Linking CXX static library libcee_wks.a
[ 43%] Built target cee_wks
make: *** [all] Error 2
Failed to build coreclr components.
```

I've researched that problem a little bit and found that the error was related somehow to libunwind-3.0 library. Any ideas how to fix it? Does it relate to the issue #872?

**Comment by naamunds** [comment]
I'm hitting the same error.  I tried back-leveling to XCode 6.3.2, and then the build succeeded, so the issue does appear to be specific to XCode 7.0.

**Comment by janvorli** [comment]
The dispatchexceptionwrapper.S uses hand-crafted __eh_frame section, so there is something in them that the new XCode doesn't like. 
This asm source contains a special helper function PAL_DispatchExceptionWrapper that is never called, but when a hardware exception like a null reference exception, division error, etc. happens, we create a special frame on stack that looks like belonging to this function so that the stack unwinder can unwind from the exception handler through this frame to the failing function and further to its callers.
By chance, I am just working on a similar thing for injecting activations into threads and I believe I can get rid of the hand-crafted __eh_frame section here and let the assembler generate it. That should fix the issue.
I will look into it.

**Comment by desunit** [comment]
Great, thanks so much - will be glad to retest it again.

**Comment by janvorli** [comment]
Fixed by #1610 - I've updated XCode on my Mac to 7.0 and verified that the build passes.

**Comment by desunit** [comment]
Thanks so much! Compiled successfully but I had to install libicu-dev first: brew install icu4c
